src/test/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilderNamespaceTest.java

```diff
@@ -0,0 +1,153 @@
+package com.google.jenkins.plugins.k8sengine;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+
+import hudson.FilePath;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.Optional;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.yaml.snakeyaml.Yaml;
+
+public class KubernetesEngineBuilderNamespaceTest {
```

Comment:
Should we test the IllegalArgumentException case for addNamespace (manifest contains multiple namespaces)?

Reply:
This no longer throws IllegalArgumentException, but now I have tests for the multiple namespace case.

Reply:
nice cleanup